pvlib/irradiance.py

```diff
@@ -2819,3 +2820,130 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
+
+
+def pvfactors_bifacial_irradiance(
+        solar_azimuth, solar_zenith, surface_azimuth, surface_tilt,
+        timestamps, dni, dhi, gcr, pvrow_height, pvrow_width, albedo,
+        n_pvrows=3, index_observed_pvrow=1,
+        rho_front_pvrow=0.03, rho_back_pvrow=0.05,
+        horizon_band_angle=15.,
+        run_parallel_calculations=True, n_workers_for_parallel_calcs=None):
+    """
+    Calculate front and back surface plane-of-array irradiance on
+    a fixed tilt or single-axis tracker PV array configuration, and using
+    the open-source "pvfactors" package.
+    Please refer to pvfactors' online documentation for more details:
+    https://sunpower.github.io/pvfactors/
+
+    Parameters
+    ----------
+    solar_azimuth: numpy array of numeric
+        Sun's azimuth angles using pvlib's azimuth convention (deg)
+    solar_zenith: numpy array of numeric
+        Sun's zenith angles (deg)
+    surface_azimuth: numpy array of numeric
+        Azimuth angle of the front surface of the PV modules, using pvlib's
+        convention (deg)
+    surface_tilt: numpy array of numeric
+        Tilt angle of the PV modules, going from 0 to 180 (deg)
+    timestamps: array of :class:datetime.datetime objects
+        List of simulation timestamps
+    dni: numpy array of numeric values
+        Direct normal irradiance (W/m2)
+    dhi: numpy array of numeric values
+        Diffuse horizontal irradiance (W/m2)
+    gcr: float
+        Ground coverage ratio of the pv array
+    pvrow_height: float
+        Height of the pv rows, measured at their center (m)
+    pvrow_width: float
+        Width of the pv rows in the considered 2D plane (m)
+    albedo: float
+        Ground albedo
+    n_pvrows: int, default 3
+        Number of PV rows to consider in the PV array
+    index_observed_pvrow: int, default 1
+        Index of the PV row whose incident irradiance will be returned. Indices
+        of PV rows go from 0 to n_pvrows-1.
+    rho_front_pvrow: float, default 0.03
+        Front surface reflectivity of PV rows
+    rho_back_pvrow: float, default 0.05
+        Back surface reflectivity of PV rows
+    horizon_band_angle: float, default 15
+        Elevation angle of the sky dome's diffuse horizon band (deg)
+    run_parallel_calculations: bool, default True
+        pvfactors is capable of using multiprocessing. Use this flag to decide
+        to run calculations in parallel (recommended) or not.
+    n_workers_for_parallel_calcs: int, default None
+        Number of workers to use in the case of parallel calculations. The
+        default value of 'None' will lead to using a value equal to the number
+        of CPU's on the machine running the model.
+
+    Returns
+    -------
+    front_poa_irradiance: numpy array of numeric
```

Comment:
Output convention could be np.array if inputs are np.array, but pandas.Series if inputs are themselves pandas objects.

Reply:
Thanks @cwhanse , as mentioned above, the decorating function should handle this request as well. Let me know what you think